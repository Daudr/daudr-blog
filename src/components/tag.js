import React, { useState } from "react"
import kebabCase from "lodash/kebabCase"

import "../styles/tag.css"
import { Link } from "gatsby"

function Tag({ tag, count }) {
  const [hover, setHover] = useState(false)

  const hoverOn = () => {
    setHover(true)
  }
  const hoverOff = () => {
    setHover(false)
  }

  return (
    <div
      style={{
        cursor: `pointer`,
        padding: `5px 10px`,
        border: `1px dashed black`,
        margin: `5px`
      }}
      className={hover ? `tag-hovered` : ``}
      onMouseEnter={hoverOn}
      onMouseLeave={hoverOff}
    >
      <Link
        style={{
          boxShadow: `none`,
          textDecoration: `none`,
          color: `inherit`,
        }}
        to={`/tags/${kebabCase(tag)}`}
      >
        {tag} {count ? `(${count})` : ``}
      </Link>
    </div>
  )
}

export default Tag
