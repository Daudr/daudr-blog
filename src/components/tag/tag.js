import React, { useState } from "react"
import kebabCase from "lodash/kebabCase"
import { Link } from "gatsby"

export const Tag = ({ tag, count, isIT = false }) => {
  const [hover, setHover] = useState(false)

  const hoverOn = () => {
    setHover(true)
  }
  const hoverOff = () => {
    setHover(false)
  }

  return (
    <Link
      style={{
        boxShadow: `none`,
        textDecoration: `none`,
        color: `inherit`,
        fontFamily: `'Anton', sans-serif`,
        textTransform: `uppercase`,
        fontWeight: `bold`,
      }}
      to={`${isIT ? "/it" : ""}/tags/${kebabCase(tag)}`}
    >
      <div
        style={{
          cursor: `pointer`,
          padding: `5px 10px`,
          border: hover ? `1px solid black` : `1px dashed black`,
          margin: `5px`,
          backgroundColor: `#FFFFFF`,
        }}
        role="button"
        tabIndex={0}
        onMouseEnter={hoverOn}
        onMouseLeave={hoverOff}
      >
        #{tag} {count ? `(${count})` : ``}
      </div>
    </Link>
  )
}

export default Tag
