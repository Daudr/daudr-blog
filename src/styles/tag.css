.tag-hovered {
    border: 1px solid black !important;
}
