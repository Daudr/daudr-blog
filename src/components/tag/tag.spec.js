import React from 'react'
import renderer from 'react-test-renderer'

import { Tag } from './tag'

describe('Tag', () => {
	it('renders correctly', () => {
		const tree = renderer.create(<Tag />).toJSON()
		expect(tree).toMatchSnapshot()
	})
})
