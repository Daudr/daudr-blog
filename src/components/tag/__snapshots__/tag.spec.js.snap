// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag renders correctly 1`] = `
<a
  href="/tags/"
  style={
    Object {
      "boxShadow": "none",
      "color": "inherit",
      "fontFamily": "'Anton', sans-serif",
      "fontWeight": "bold",
      "textDecoration": "none",
      "textTransform": "uppercase",
    }
  }
>
  <div
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "border": "1px dashed black",
        "cursor": "pointer",
        "margin": "5px",
        "padding": "5px 10px",
      }
    }
  >
    #
     
    
  </div>
</a>
`;
